Object Replication for Get Properties and Download Response

amnguye opened the pull request.

amnguye committed:
Adding OR header parsing to Get Properties and Download Response

Saving new OR changes

Addressing PR comments, still needs work on Test Recordings

Updated test recordings

amnguye committed:
Adding OR header parsing to Get Properties and Download Response

Saving new OR changes

Addressing PR comments, still needs work on Test Recordings

Updated test recordings

amnguye committed:
Adding OR header parsing to Get Properties and Download Response

Saving new OR changes

Addressing PR comments, still needs work on Test Recordings

Updated test recordings

amnguye committed:
Adding OR header parsing to Get Properties and Download Response

Saving new OR changes

Addressing PR comments, still needs work on Test Recordings

Updated test recordings

seanmcc-msft commented:
You need to record the new unit tests to pass CI.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:
```diff
@@ -2424,6 +2426,72 @@ private void SetNameFieldsIfNull()
                         .ConfigureAwait(false);
+                    if (response.Value.ObjectReplicationRuleStatus.Count > 0)
+                    {
+                        IDictionary<string, string> ParsedRuleIdStatus = new Dictionary<string, string>();
```
I think we should put this logic in BlobExtensions.cs (might not exist in this branch, see https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs as an example.

seanmcc-msft replied:
Also, I think we should rename this ```ObjectReplicationRules```.

ShivangiReja committed:
[Service Bus] Design and implement rule manager (#11428)

 [Service Bus] Design and implement rule manager

JoshLove-msft committed:
Include tracking ID in exceptions (#11391)

JoshLove-msft committed:
Add Send IEnumerable (#11427)

jsquire committed:
[Event Hubs Client] Assorted Small Improvements (#11412)

The focus of these changes is to collect a set of smaller miscellaneous
changes to improve performance and ensure data integrity.  The prominent
changes are:

  - Centralized an empty read-only set of events to be returned when an
    attempt to read finds no events in the partition; previously, a new set
    was allocated for each call.

  - Making a shallow copy of an event when added to a batch to ensure that
    changes made to the event's property bag while the event is in a batch
    does not result in the ability to exceed the size of the batch.

jsquire committed:
[Event Hubs Client] Live Test Improvements (#11421)

The focus of these changes is to unify the timeout value applied as a
safety measure to prevent hangs, hooking it into the test environment to
allow for configuration to differ between the local and CI environments.

A few minor tweaks were also made to attempt to stablize tests that have
been misbehaving during nightly runs.

pakrym committed:
Add role assignment to README.md (#11455)

kinelski committed:
[FormRecognizer] Preview 1: Removed _cancellationToken field from Operation subclasses (#11465)

maririos committed:
confidence and more (#11467)

jsquire committed:
[Cross-Service Samples] IoT Hub => Event Hub Connection String Translation (#11314)

The focus of these changes is to introduce a new sample that illustrates how to
query IoT Hub for the connection string to its "built-in Event Hub endpoint" used
for sharing device telemetry.

heaths committed:
Fix Key Vault live tests (#11415)

Occasional issue with live tests caused by test assemblies running
concurrently. We need to skip managed keys and secrets (only secrets was
actually failing, but keys could as well) since certificates may be
getting deleted simultaneously.

kinelski committed:
[FormRecognizer] Preview 1: throw RequestFailedException when long-running operation has failed (#11460)

pakrym committed:
Add TestEnvironment (#11410)

heaths committed:
De/serialize pipe-delimited array of RegexFlags  (#11461)

* De/serialize |-delimited array of RegexFlags

Fixes #10748

* Declare related read-only collection properties

* Update public APIs

* Resolve PR feedback

jsquire committed:
[Event Hubs Client] Test Environment Housekeeping (#11479)

The focus of these changes is to clean up some inconsistencies caused by
dueling changes involving the test environment, and reformat/refactor some
of the recent work into something that closer matches the internal styles used
within the Event Hubs source.

annelo-msft committed:
.NET FormRecognizer README and initial samples. (#11389)

* README and first sample

* add 1st snippets

* adding recognize samples

* troubleshooting

* add section header for package installation

* add snippets to README

* add Next Steps section

* tweaks

* add links

* pr fb

* nit

* update snippets

* merge test environment changes

* pr fb

* pr fb

ellismg committed:
[EventHubs] Enable the FxCop Analysizers (#11423)

As a follow up to #11414, I was trying to understand why did not see
errors from the "FxCop Analyiszers", which would have pointed out that
issue to us.

The reason was simple - we had disabled the analysis for these projects.

I've re-enabled them, either fixing up issues or adding suppressions.
We ended up catching some reasonable errors (a few cases where we called
the ArgumentException constructor but messed up the order of the message
and the parameter name) as well as a case where we were passing two
parameters to an internal method but then disregarding them. In
practice, the code we had was still "correct", because it happened to
only be called from a single place today and the only caller was passing
the same values that the implementation was actually using.

Overall, I think taking this would be worthwhile. It brings this library
back in line with the the others.  I could imagine some library wide
supressions of rules here, specifically the one about not calling
virtual methods from constructors, since in practice we are unlikely to
hit the problems the rule intends to guard for, but for now I've just
made the diff look "as bad as possible" by doing supressions around the
offending lines with justifications.

maririos committed:
Confidence fix and RecognizeCustomForm test (#11481)

* changes

* overwrite serialization

bashahee committed:
Update Information Protection SDK Clients (#11464)

* Create security_resource-manager.txt

* Move txt file

* Update clients

* Update versions

Kotasudhakarreddy committed:
[AppServie]:Azure .Net SDK update for PowerShell (#11442)

* Azure .Net SDK update for PowerShell

* Added _metadata file

* Generated using latest version of AutoRest

christothes committed:
Create initial ci for Azure.Data.Tables (#11502)

* create initial ci for Azure.Data.Tables

* Azure.Data.Table => Azure.Data.Tables

Co-Authored-By: Pavel Krymets <pavel@krymets.com>

* Azure.Data.Table => Azure.Data.Tables

Co-authored-by: Pavel Krymets <pavel@krymets.com>

kinelski committed:
[FormRecognizer] Preview 1: fix modelId not being set in operation (bug fix) (#11477)

AlexanderSher committed:
Two small bugfixes for Azure.Identity (#11504)

* Two small bugfixes

* Rename method

maririos committed:
Add changelog to FR (#11507)

* changelog

* fix typos

* Update sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md

Co-Authored-By: Anne Thompson <annelo@microsoft.com>

* Update sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md

Co-Authored-By: Anne Thompson <annelo@microsoft.com>

Co-authored-by: Anne Thompson <annelo@microsoft.com>

AlexanderSher committed:
Fix #8624: Unable to delete the key when test completed (#11205)

christothes committed:
Initial generated Azure.Data.Tables client (#11506)

* initial commit of basic structure for Azure.Data.Tables

* add tests.yml

* fix readme

* fixup naming and package version

* ci merge conflict

* add /sdk/tables/ to codeowners

* make canary test work

* create changelog entry

* # Azure Tables client *library* for .NET

* export api

kinelski committed:
[FormRecognizer] Preview 1: Add docstrings to the Form Recognizer library (#11496)

pakrym committed:
Fix FR test-resources template and New-TestResources script (#11512)

christothes committed:
Adding minor context to the Public API additions section (#11513)

* Adding minor context to the Public API additions section

* adding the /api dir

annelo-msft committed:
Finishing PR 11491: Samples (#11515)

* sample docs

* PR feedback

* disable samples

* missed rename

* fix build

* sample updates

* fix PR checks

* More tweaks

* update snippets

Co-authored-by: Mariana Rios Flores <mariari@microsoft.com>

blueww committed:
[Storage] Support File share softdelete, accesstier, NFS, and TagFilter in ManagementPolicy (#11500)

maririos committed:
small changes (#11523)

danielmarbach committed:
[Azure Service Bus] Replace Stopwatch with ValueStopwatch (#11266)

* Introduce ValueStopwatch

* Move namespace

* Original license

* Replace Stopwatch with ValueStopwatch where needed

* Moved to core

* Used ValueStopwatch from Core

* Address Review comments

jsquire committed:
[Samples] Iot Hub => Event Hubs Connection String Translation (#11537)

The focus of this change is to adjust the API version of the ARM template
for better compatibility in non-public clouds.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs:
```diff
@@ -369,4 +410,304 @@ public static partial class BlobsModelFactory
+    /// <summary>
+    /// BlobProperties
+    /// </summary>
+    public partial class BlobProperties
```
Let's pull this out into it's own file.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs:
```diff
@@ -369,4 +410,304 @@ public static partial class BlobsModelFactory
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID.
+        /// </summary>
+        /// <returns></returns>
+        internal IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds()
```
Is this the same logic as `BlobDownloadDetails.ParseObjectReplicationIds()`?  If so, I think we should consolidate it and move it to `BlobExtensions`.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs:
```diff
@@ -369,4 +410,304 @@ public static partial class BlobsModelFactory
+        /// Creates a new BlobProperties object backed by BlobPropertiesInternal
+        /// </summary>
+        /// <param name="properties">The BlobPropertiesInternal returned with the request</param>
+        public BlobProperties(BlobPropertiesInternal properties)
```
Maybe make this an extension method on `BlobPropertiesInteral`?  Either way is fine, but most of the model conversions in Data Lake use extension methods.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/swagger/readme.md:
````diff
@@ -4,7 +4,7 @@
 ``` yaml
 # Generate blob storage
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/storage-dataplane-preview/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-12-12/blob.json
+input-file: C:\azure-rest-api-specs\specification\storage\data-plane\Microsoft.BlobStorage\preview\2019-12-12\blob.json
````
Remember to revert this.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/swagger/readme.md:
````diff
@@ -1447,6 +1447,15 @@ directive:
     $.BlobExpiryOptions["x-az-public"] = false;
 ```
 
+### Make BlobPropertiesInternal internal
````
Oh nice, you already made BlobPropertiesInternal internal.  Run `sdk/storage/generate.ps1` so this takes effect in the generated code, and  `eng/scripts/Export-API.ps1` so the change takes effect in `api/Azure.Storage.Blobs.netstandard2.0.cs`

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:
```diff
@@ -429,6 +429,34 @@ async Task Verify(Response<BlobDownloadInfo> response)
+            try
+            {
+                await source_blob.UploadAsync(new MemoryStream(GetRandomBuffer(Constants.KB)));
+                await Task.Delay(300000);
```
What are we waiting for?  Also, Ted prefers we use `https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs#L338`

amnguye replied:
It takes at least 5 minutes for the replication to occur. If we try to get the properties, it will not show up. (I will use add the delay instead in the next commit)

rickle-msft replied:
What I ended up doing was just hard coding a blob name and only doing getProperties on it instead of uploading. Since it's only recorded, it'll always be the same and not have any issues with collisions, and then you don't have to wait at all.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:
```diff
@@ -429,6 +429,34 @@ async Task Verify(Response<BlobDownloadInfo> response)
+        [Test]
+        public async Task DownloadAsync_ObjectReplication()
+        {
+            // This is a recorded ONLY test with a special container we previously setup, as we can't auto setup policies yet
```
Are ORS policies persistent?  Like if I create an ORS policy, it's not going to go away unless I delete it, correct?

seanmcc-msft replied:
Also, is it only possible to apply an ORS policy to a container via the management plane?

amnguye replied:
ORS are policies are persistent. They can only be set and deleted through the management plane

seanmcc-msft replied:
Has ORS already been added to the management plane SDK?

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:
```diff
@@ -1863,6 +1891,34 @@ public async Task GetPropertiesAsync_AccessConditionsFail()
+            try
+            {
+                await source_blob.UploadAsync(new MemoryStream(GetRandomBuffer(Constants.KB)));
+                await Task.Delay(300000);
```
Same as other `Task.Delay()` comment

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs:
```diff
@@ -34,7 +34,7 @@ public abstract class BlobTestBase : StorageTestBase
 
         public BlobTestBase(bool async, BlobClientOptions.ServiceVersion serviceVersion, RecordedTestMode? mode = null)
-            : base(async, mode)
+            : base(async, RecordedTestMode.Live)
```
Remember to revert this.

danielmarbach committed:
Use ValueStopWatch in EventHub AMQP (#11530)

chidozieononiwu committed:
Fix Error in Update-PkgVersion (#11539)

jsquire committed:
[Smoke Tests] Sample: IoT Hub Connection String Translation (#11388)

The focus of these changes is to adjust the base infrastructure for the
smoke tests and extend it to allow for integration of the cross-service samples.

The sample demonstrating querying IoT Hub for an Event Hubs connection string
has been integrated as part of the test suite.

seanmcc-msft committed:
Feature/storage/data lake rename sas bug (#11290)

pakrym committed:
Pass all recorded variables through sanitizer (#11541)

heaths committed:
Return pageables from SearchServiceClient.GetIndexes(Async) (#11517)

Fixes #11053

pakrym committed:
Fix recordings when array is the root of the service response (#11543)

isra-fel committed:
Update core ci so mgmt plane PRs do not trigger it (#11495)

pakrym committed:
Improve template to include tests, live resources, recordings… (#11540)

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs:
```diff
@@ -0,0 +1,280 @@
+        /// <summary>
+        /// Returns the date and time the blob was last modified. Any operation that modifies the blob, including an update of the blob's metadata or properties, changes the last-modified time of the blob.
+        /// </summary>
+        public System.DateTimeOffset LastModified { get; internal set; }
```
I think we might be able to do
```
LastModified => _properties.LastModified;
CreatedOn => _properties.CreatedOn;
etc.
```
And then just take a BlobPropertiesInternal in the constructor.

amnguye replied:
I did try to do this, but in ToBlobProperies method in BlobExtension I ran into an error about these values being readonly, when we are attempting to set them.

seanmcc-msft replied:
Oh gotcha.  Sorry I gave conflicting advise on this one, lets leave it as-is.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:
```diff
@@ -621,6 +624,51 @@ public partial class BlobMetrics
     public partial class BlobProperties
     {
         public BlobProperties() { }
```
Let's make both these constructors internal.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:
```diff
@@ -1863,6 +1886,29 @@ public async Task GetPropertiesAsync_AccessConditionsFail()
+        public async Task GetPropertiesAsync_ObjectReplication()
+        {
+            // This is a recorded ONLY test with a special container we previously setup, as we can't auto setup policies yet
+            var source_container = InstrumentClient(new BlobContainerClient(TestConfigDefault.ObjectReplicationSource, "test1"));
```
Let's switch these variable names to camel case.

seanmcc-msft commented on sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:
```diff
@@ -1863,6 +1886,29 @@ public async Task GetPropertiesAsync_AccessConditionsFail()
         }
 
+        [Test]
+        public async Task GetPropertiesAsync_ObjectReplication()
```
Let's add TODOs on these tests to figure out what we want to do with handling setting ORS on containers.

seanmcc-msft commented:
@amnguye , merge the 73 base branch into your branch.  The merge conflict aren't too bad.

seanmcc-msft commented:
@rickle-msft @kasobol-msft , can you take a look?  This PR is blocking Jumbo Blobs and Quick Query.

amnguye commented:
> @amnguye , merge the 73 base branch into your branch. The merge conflict aren't too bad.

Just made the merge

sandeepkv93 committed:
Simulate Eviction API cameCase fix for Spot Virtual Machines and VMScaleSet VMs (#11558)

Co-authored-by: Sandeep Vishnu <savishnu@microsoft.com>

isra-fel committed:
unblock test projec (#11561)

christothes committed:
Create initial test-resources.json for tables (#11538)

* create initial test-resources.json

* added a recorded test

* pr fixes

* inline table uri

pakrym committed:
Add guardrails to TestEnvironment (#11570)

christothes committed:
Pull the 2019-02-02 swagger (#11572)

* pull the 2019-02-02 swagger

* fixup options and link to specific commit

* update exports

AlexanderSher committed:
Handle JsonException when process returns non-json output (#11575)

pakrym committed:
Relax guardrails in TestEnvironment when used outside of recorded test (#11579)

seanmcc-msft committed:
Merge branch 'feature/storage/stg73base' into feature/storage/object-replication

pakrym committed:
Add permission info to Azure.Extensions.AspNetCore.DataProtection.Keys/README.md (#11576)

AlexGhiondea committed:
Add attribution for .NET Core's Apache license (#11379)

* Add attribution for .NET Core's Apache license

* Update NOTICE.txt

Include the whole license text.

isra-fel committed:
Update ci.yml for Azure Pipelines (#11609)

exclude `eng/mgmt/` in pr validation

erich-wang committed:
Enable to playback track 1 session record for track 2 mgmt sdk test cases (#11535)

amnguye committed:
Rerecorded tests with testsconfig changes; version tests, some SAS tests, abort copy and oauth tests not added

amnguye committed:
Rerecorded tests with testsconfig changes; version tests, some SAS tests, abort copy and oauth tests not added

amnguye committed:
Rerecorded tests with testsconfig changes; version tests, some SAS tests, abort copy and oauth tests not added

pakrym committed:
Change ProtectKeysWithAzureKeyVault to take a Uri (#11588)

amnguye committed:
Rerecorded Encrypton Scope tests

pakrym committed:
Normalize special headers in playback record matcher (#11591)

christothes committed:
Support continuation tokens for tables (#11620)

* support continuationtokens for tables

* update api

* comment swagger and generator workarounds, move those files out of /swagger

* break up the tests, re-record

ShivangiReja committed:
[Service Bus] Improve errors and update API docs as per the feedback from UX study (#11592)

[Service Bus] Improve errors and update API docs as per the feedback from UX study

nourhanFarid committed:
A nofari/luis sdk (#11596)

* adding LuisApp import/ export lu format, adding LuisApp import v2 app, adding azure account related auth using arm token

* fixing import tests

* fixing export return type from void to string

* adding query parameter export extensoin in app version exporting

* fixing in query parameters for export app to have a const value "lu"

* fising export retun type to file

* fixing export return type to file

* Adding tests for import app/appVersion lu format, export appVersion lu format, import app/appVersion v2

* reverting auth key

* reverting changes made to global app id

* adding missing sessions

* changing from lu extension to .txt extension

* reverting email

* changing input file extension

* bumping version

* reverse undesired changes for property group

* removing not needed urls

* removing email and updating test

* updating tests

* changes in eror return response

* fixing input spelling

* adding CHANEGLOG.md file

* fixing test

JoshLove-msft committed:
Sender updates (#11580)

MiYanni committed:
AutoRest version update (#11621)

* Updated version. Removing shared files and then copying them again to fix folder name casing.

* Re-added shared files with fixed folder name.

* Regeneration after version update.

* Copied wrong version of the file.

* Update generator version and add string request fix.

* Fix path folder casing.

* Fix casing.

* More casing.

* Last of casing. Now, it is fixed for the repo.

heaths committed:
Set appropriate Accept header for Search clients (#11562)

* Set appropriate Accept header for Search clients

Fixes #11532

* Resolve PR feedback

* Unconditionally change the Content-Length in sanitizer

isra-fel committed:
maintain mgmt project list (#11613)

JoshLove-msft committed:
Processor updates (#11587)

rickysun93 committed:
Add azurefile/prefix/fileFilter/partitionRootPath/zipFile to ADF swagger (#11603)

Co-authored-by: Haoran Sun <haorsun@microsoft.com>

sivanguetta committed:
Monitor- metricAlert_API- Fix WebtestLocationAvailabilityCriteria criteria type (#11615)

romahamu committed:
[Storagecache] Generate SDK for New API version 2020-03-01 (#11618)

* Add newly recorded tests

* Add src code related to new API vesrion 2020-03-01

* Change for API version handling in the tests.

* For 2020-03-01 API cache identity will always be SystemAssigned by default and for older versions it will set to None.

* Add default settings for disk encryption and cache identity features for a cache.

* Changes to cache test to verify default cache identity and security settings.

* Formatting fixes

* Use centraluseuap region.

* Update metadata file.

* Update API tag to 2020-03-01

* Version bump to 1.1.0 for new API 2020-03-01 of SDK.

gautamd-ms committed:
Updating release version for maintenance sdk (#11628)

pakrym committed:
Compare bodies in playback tests (#11624)

danielmarbach committed:
[Service Bus] Change from static factory method to constructor for copying received message (#11641)

* Ability to create a ServiceBusMessage from a ServiceBusReceivedMessage

* Guard against null

pakrym committed:
Remove sync over async in response content stream (#11643)

kinelski committed:
[FormRecognizer] Post-release: updating project version number (#11549)

seanmcc-msft committed:
Added BaseBlobClient.CopyFromUri() (#11635)

annelo-msft committed:
Add Code Owners for Form Recognizer (#11647)

* add code owners for form recognizer

* bug fix

* whitespace

pakrym committed:
Add TestEnvironment information to Test Framework doc (#11547)

* Add TestEnvironment information to Test Framework doc

* Link!

pakrym committed:
Update Autorest C# (#11652)

heaths committed:
Define onlyIfUnchanged parameter for CreateOrUpdate and Delete operations (#11420)

* Combine MatchConditions and SearchRequestOptions

Fixes #11052

* Redefine ETag properties as Azure.ETag?

Fixes #11385

* Update public APIs

* Define onlyIfUnchanged for ETag support (#2)

This was suggested to match what AppConfiguration does. It simplifies it, with no practical reason for IfNoneMatch. We could always add that later as well.

* Update CHANGELOG and public APIs

amnguye committed:
Revert rerunned tests, rerecorded OR tests

amnguye committed:
Revert rerunned tests, rerecorded OR tests

amnguye committed:
Revert rerunned tests, rerecorded OR tests

seanmcc-msft commented on sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurations.cs:
```diff
@@ -128,13 +140,24 @@ public class TestConfigurations
+        /// <summary>
+        /// Gets the source tenant to use for any tests related to Object Replication.
+        /// </summary>
+        public static TenantConfiguration DefaultObjectReplicationDestinationTenant =>
```
I thought we were going to take a dependency on the management plane SDK to create ORS policies and assign them, instead of adding a new TenantConfiguration.

northtyphoon committed:
Update container registry sdk using latest spec (#11625)

* update container registry sdk from latest spec

* Update Microsoft.Azure.Management.ContainerRegistry.csproj

Co-authored-by: Yeming Liu <Yeming.Liu@microsoft.com>

jsquire committed:
[Event Hubs Client] Publish Events Without Batch (#11651)

The focus of these changes is to allow the publishing of events without the
overhead of creating an explicit batch.  In this scenario the caller assumes
the risk with respect to exceeding the maximum size allowed by the Event Hubs
service, trading safety for lower overhead and higher throughput.

jsquire committed:
[Event Hubs Client] Misc. Formatting and Test Tweaks (#11648)

The focus of these changes is to make a miscellaneous set of small tweaks and
fixes, largely focused on formatting and tests.

christothes committed:
Clarify use of EnableApiCompat and its relation to /api/netstandard2.0.cs (#11655)

* clarify use of EnableApiCompat and its relation to the netstandard2.0.cs file

* update wording

vinagesh committed:
Adding support for GenerateId max length (#11659)

kinelski committed:
[FormRecognizer] Enable test recordings (#11654)

danieljurek committed:
add ServiceDirectory to usages of remove-test-resources.yml (#11666)

VeryEarly committed:
Applicationinsights 0.3.0 (#11694)

* delete outdated generate.cmd

* add generate.ps1

* generate SDK from updated swagger specs

* Update generate.ps1

Co-authored-by: Yeming Liu <Yeming.Liu@microsoft.com>

filizt committed:
[Blueprints] Remove resource group name and location length validation in a blueprint (#11555)

* remove resource group name and location length validation in a blueprint

* Address feedback

Co-authored-by: Filiz Topatan <lsefitopata@microsoft.com>

VeryEarly committed:
Operationalinsights 0.20.0 (#11698)

* remove generate.cmd for operationalinsights

* add generate.ps1 for operationalinsights

* generate SDK from updated swagger spec

* update release info

* Update generate.ps1

Co-authored-by: Yeming Liu <Yeming.Liu@microsoft.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#11636)

pakrym committed:
Prepare for Azure.Core 1.2.1 release (#11705)

soeom committed:
Regenerate CosmosDB SDK (#11542)

* generated sdk

* Fix for long running operations

* fix per comment

* change version number in assemblyinfo

Co-authored-by: Meha Kaushik <mekaushi@microsoft.com>

bgklein committed:
Fix Batch Tests to work with 3.1 (#11688)

* Force clean history

* minor

pakrym committed:
Update Autorest C# (#11687)

pakrym committed:
Update Azure.Core version to 1.2.1 (#11709)

seanmcc-msft committed:
Added DataLakeFileClient.Upload() overload that allows setting metada… (#11622)

jsquire committed:
[Event Hubs Client] Test Timing Tweaks (#11710)

The focus of these changes is to fix some incorrect timeout values for test
cancellation which potentially contributed to an aborted run due to hanging in
CI.  Some of the synchronization points for partition processing were also
adjusted to hook a more deterministic point and, hopefully, improve reliability.

heaths committed:
Rename Search models for Preview 3 (#11707)

* Rename RegexFlags to RegexFlag

Fixes #11510

* Rename various Search models

Fixes #11243

* Update public APIs

Also fixed a couple other minor renames/accessibility changes from APIView.

* Update public APIs

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#11713)

adrianfj committed:
Parse expires_on with an invariant culture and no datetime style when ManagedIdentityClient tries to deserialize authentication response (#11563)

* Update ManagedIdentityClient.cs

Parse expires_on with an invariant culture and no datetime style when ManagedIdentityClient tries to deserialize authentication response.

* Updated ManagedIdentityCredentialTests.cs

Changed mocked responses to return expires_on in an invariant culture.

JoshLove-msft committed:
Create TestFramework project (#11663)

ShivangiReja committed:
Make ServiceBusRuleManager and filters internal (#11719)

seanmcc-msft committed:
Merge branch 'master' into feature/storage/stg73base

christothes committed:
Support additional types in WriteObjectValue (#11706)

danielmarbach committed:
[Event Hub] Use ExceptionDispatchInfo consistently accross AqmpClient and Producer like in AmqpConsumer (#11727)

* Align AmqpClientand AmqpProducer with AmqpConsumer

* Unnecessary semicolon

jsquire committed:
[Event Hubs Client] Ignore Hanging Test (#11732)

The focus of these changes is to mark a test that has been intermittently
hanging in CI runs to be ignored, pending deeper investigation.

chidozieononiwu committed:
Update ubuntu VMImage (#11545)

* Update ubuntu vmimage

* Adding Mikes Bypass DNS server bugfix as a step

christothes committed:
Implement remaining table entity operations (#11715)

* implement remaining table entity operations

* re-record tests

* pr comments

* fix some random values for recordings

* make dateTimes UTC

* correct universalTime for entity dates

* it helps to check in the new recordings :)

* I think the dates are fixed now

mikeharder committed:
Fix path to bypass-local-dns.yml

heaths committed:
Prevent base classes from being instantiated (#11720)

* Prevent base classes from being instantiated

Works around #9686

* Update public APIs

heaths committed:
More Search model renames (#11739)

* More Search model renames

Fixes #11738 and picks up a few description improvements in swagger.

* Update public APIs

danielmarbach committed:
[Service Bus] Avoid rethrowing exception with throw ex syntax when exception unchanged (#11679)

* Try to not mess with the original exception if returned

* Apply suggestion from code review of jesse

pakrym committed:
Add support for test environment files (#11673)

* Add support for storing the test environment into the file

* Updates

* Consumption

* Update TestEnvironment.cs

allenjzhang committed:
Management SDK project template and build infrastructure update (#11677)

* 1. Added management SDK client project template, 2. Adjusted Build.Data to support new management SDK projects

* 1. Addressed comments: moved warning to data.prop, updated initial version, remove extra line
2. added empty tags if none specified
3. added flag to copy ci.yml and test-resource.json into parent folder

* Added link to the dotnet template in CONTRIBUTING.md

* Update minimal test-resource ARM template file

allenjzhang committed:
Fixing readme file

allenjzhang committed:
Update README.md

tg-msft committed:
Rename SearchIndexClient to SearchClient (#11748)

amnguye committed:
Updated tests using Management SDK to create policy on accounts

amnguye committed:
Updated tests using Management SDK to create policy on accounts

JoshLove-msft committed:
Add TestFramework proj to Track 2 sln files (#11747)

seanmcc-msft committed:
Merge branch 'master' into feature/storage/stg73base

jaschrep-msft committed:
Feature/storage/jumbo blob (#11548)

* StorageTransferOptions has long properties

* Checkpoint

* Pooled Memory Stream functional

* Jumbo Blobs testing complete

* Some work for pulling blobs partitioned uploader to common removed for
later work.

* Addressed comments

* Export API

* Bug fix and explicit tests

* Addressed minor comments

* Versioning of block blob limit properties

* Renamed new properties

* export api

* Reverted some of the previous renames

amnguye committed:
Merge branch 'feature/storage/object-replication' of https://github.com/amnguye/azure-sdk-for-net into feature/storage/object-replication

amnguye commented:
Give me a second I think I broke my branch somehow.

amnguye committed:
Merge branch 'feature/storage/object-replication' of https://github.com/amnguye/azure-sdk-for-net into feature/storage/object-replication

amnguye committed:
Removing unnecessary added merges
